include/cgimap/api06/changeset_upload/osmobject.hpp

```diff
@@ -97,9 +97,9 @@ namespace api06 {
 
     osm_nwr_signed_id_t id() const { return *m_id; }
 
-    bool has_changeset() const {  return ((m_changeset) ? true : false); }
-    bool has_id() const { return ((m_id) ? true : false); };
-    bool has_version() const { return ((m_version) ? true : false); }
+    bool has_changeset() const {  return (m_changeset ? true : false); }
```

Comment:
maybe we could use `m_changeset.has_value()`  and get rid of `? true : false`. Same for the next two lines.

Reply:
yup, changed to `has_value()`